lib/filterx/func-str.c

```diff
@@ -497,3 +519,198 @@ filterx_function_includes_new(FilterXFunctionArgs *args, GError **error)
   return _function_affix_new(args, "includes", _function_includes_process, FILTERX_FUNC_INCLUDES_USAGE,
                              error);
 }
+
+static FilterXObject *
+_strcasecmp_eval(FilterXExpr *s)
+{
+  FilterXStrcasecmp *self = (FilterXStrcasecmp *) s;
+  FilterXObject *result = NULL;
+  FilterXObject *a_obj = NULL, *b_obj = NULL;
+
+  const gchar *a_str = NULL;
+  if (!self->a_literal)
+    {
+      a_obj = filterx_expr_eval(self->a.expr);
+      if (!a_obj)
+        goto exit;
+
+      gsize a_str_len = 0;
+      if (!filterx_object_extract_string_ref(a_obj, &a_str, &a_str_len))
```

Comment:
This might return a non NUL terminated string from an indirect LogMessage NVPair's Value. Passing it to a  function that expect it to be NUL terminated is not ideal. We should `APPEND_ZERO` it, right?

Reply:
The same is available in glib, but we need to calculate the MIN(a,b) then.

Reply:
Fixed.

Reply:
Thank you.